backend/shared/src/update-group-contracts-internal.ts

```diff
@@ -55,12 +58,21 @@ export async function addGroupToContract(
       })
   }
 
-  if (group.id === NON_PREDICTIVE_GROUP_ID && !contract.nonPredictive) {
+  if (group.id === UNRANKED_GROUP_ID && !contract.isRanked) {
     await firestore.collection('contracts').doc(contract.id).update({
-      nonPredictive: true,
+      isRanked: false,
     })
     if (recordEdit) {
-      await recordContractEdit(contract, recordEdit.userId, ['nonPredictive'])
+      await recordContractEdit(contract, recordEdit.userId, ['isRanked'])
+    }
+  }
+
+  if (group.id === UNSUBSIDIZED_GROUP_ID && !contract.isSubsidised) {
```

Comment:
Or is it the fact I am just putting both these IF statements for unsubsidized and unranked in the same function and need to create 2 functions?

Reply:
hmm that doesn't seem to be it either tho

Reply:
I think you just have your cases backwards.